src/test/java/seedu/address/logic/LogicManagerTest.java

```diff
@@ -60,12 +66,39 @@ public void getFilteredExpenseList_modifyList_throwsUnsupportedOperationExceptio
         logic.getFilteredExpenseList().remove(0);
     }
 
-
     @Test
     public void getExpenseStats_returnsEmptyMapWhenNoEntries() throws NoUserSelectedException {
         assertTrue(logic.getExpenseStats().size() == 0);
     }
 
+    @Test
+    public void getExpenseStatsReturnsMapWithCorrectEntries() throws NoUserSelectedException {
+        Expense validExpense = new ExpenseBuilder().build();
+        model.addExpense(validExpense);
+        model.updateStatsMode(StatsMode.DAY);
+        LinkedHashMap<String, Double> map = logic.getExpenseStats();
+        assertTrue(map.size() > 0);
+        assertTrue(map.containsKey(validExpense.getDate().toString()));
+        assertTrue(map.get(validExpense.getDate().toString()) == validExpense.getCost().getCostValue());
+
+        model.updateStatsMode(StatsMode.MONTH);
+        map = logic.getExpenseStats();
+        String month = new SimpleDateFormat("MMM-YYYY").format(validExpense.getDate().fullDate.getTime());
+        assertTrue(map.size() > 0);
+        assertTrue(map.containsKey(month));
+        assertTrue(map.get(month) == validExpense.getCost().getCostValue());
+
+        model.deleteExpense(validExpense);
```

Comment:
Dont delete the expense at the end. Use a `@BeforeEach` annotated method instead to reset the model

Reply:
ok done